qemu/tests/cdrom.py

```diff
@@ -335,18 +342,52 @@ def get_device(vm, dev_file_path):
                 raise error.TestFail("Could not find a valid cdrom device")
         return device
 
-    def get_testing_cdrom_device(session, cdrom_dev_list):
+    def get_match_cdrom(session, serial_num):
+        """
+        Find the cdrom in guest which is corresponding with the CML
+        according to the serial number.
+
+        :param session: VM session.
+        :param serial num: serial number of the cdrom.
+        :return match_cdrom: the cdrom in guest which is corresponding
+                             with the CML according to the serial number.
+        """
+        error.context("Get matching cdrom in guest", logging.info)
+        show_serial_num = "ls -l /dev/disk/by-id"
+        serial_num_output = session.cmd_output(show_serial_num)
+        if serial_num_output:
+            for line in serial_num_output.splitlines():
+                if utils_misc.find_substring(str(line), str(serial_num)):
+                    serial_cdrom = line.split(" ")[-1].split("/")[-1]
+                    break
+            if not serial_cdrom:
+                raise error.TestFail("Could not find the serial number"
+                                     "which is same in Qemu CML.")
+
+        show_cdrom_cmd = "ls -l /dev/cdrom*"
+        dev_cdrom_output = session.cmd_output(show_cdrom_cmd)
+        if dev_cdrom_output:
+            for line in dev_cdrom_output.splitlines():
+                if utils_misc.find_substring(str(line), str(serial_cdrom)):
+                    match_cdrom = line.split(" ")[-3]
+                    return match_cdrom
+            raise error.TestFail("Could not find the corresponding cdrom"
+                                 "in guest which is same in Qemu CML.")
+
+    def get_testing_cdrom_device(session, cdrom_dev_list, serial_num=None):
         """
         Get the testing cdrom used for eject
         :param session: VM session
         :param cdrom_dev_list: cdrom_dev_list
         """
-        if params["os_type"] == "windows":
-            winutil_drive = utils_misc.get_winutils_vol(session)
-            winutil_drive = "%s:" % winutil_drive
-            cdrom_dev_list.remove(winutil_drive)
         try:
-            testing_cdrom_device = cdrom_dev_list[-1]
+            if params["os_type"] == "windows":
+                winutil_drive = utils_misc.get_winutils_vol(session)
+                winutil_drive = "%s:" % winutil_drive
+                cdrom_dev_list.remove(winutil_drive)
+                testing_cdrom_device = cdrom_dev_list
```

Comment:
this code is only removing winutils ,we might still get a list of cdrom 
pls consider use cdrom_dev_list[-1] or use serial for windows as well.